requirements/ci/py38.yml

```diff
@@ -31,6 +34,7 @@ dependencies:
   - nc-time-axis
   - pandas
   - pip
+  - pykdtree
```

Comment:
This should also be included in [requirements/all.txt](https://github.com/SciTools/iris/blob/e74356961f6d6d944c16503392eefb596bb548b5/requirements/all.txt)